superdesk/core/auth/privilege_rules.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, cast
+
+from quart_babel import gettext
+from superdesk.errors import SuperdeskApiError
+from superdesk.users.async_service import get_privileges, is_admin
+from superdesk.core.types import HTTP_METHOD, AuthRule, Request
+
+
+def required_privilege_rule(privilege: str) -> AuthRule:
+    """
+    Creates an authentication rule that restricts access based on a specific privilege.
+
+    Args:
+        privilege (str): The privilege required to access the resource.
+
+    Returns:
+        AuthRule: An asynchronous rule function that checks if the user has the required privilege.
+
+    The rule verifies:
+    - If the user is an admin, access is granted without further checks.
+    - Otherwise, the user's privileges are retrieved and checked against the required privilege.
+
+    Raises:
+        SuperdeskApiError: If the user does not have the required privilege.
+    """
+
+    async def internal_rule(request: Request):
+        current_user = request.user
+
+        # TODO-ASYNC: Check if for admin we need to pull all privileges instead
+        if is_admin(current_user):
+            return None
```

Comment:
This should be fine, if the user is admin then they have all privileges by design